### ISSUE TITLE
[BUG] Azure Data Factory reflect AttributeError SubscriptionClient has no attribute subscriptions

### ISSUE THREAD

**Issue by alapinsk** [databrickslabs/dbx#127]
## Steps to Reproduce (for bugs)
Running command 
 ```dbx datafactory reflect```
results in ```AttributeError: 'SubscriptionClient' object has no attribute 'subscriptions'```
 

## Stack trace
```
Traceback (most recent call last):
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\runpy.py", line 193, in _run_module_as_main      
    "__main__", mod_spec)
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\Scripts\dbx.exe\__main__.py", line 7, in <module>    
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\site-packages\click\core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\site-packages\click\core.py", line 1053, in main 
    rv = self.invoke(ctx)
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\site-packages\click\core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\site-packages\click\core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\site-packages\click\core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\site-packages\click\core.py", line 754, in invoke    return __callback(*args, **kwargs)
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\site-packages\dbx\commands\datafactory.py", line 
70, in reflect
    reflector = DatafactoryReflector(specs_file, subscription_name, resource_group, factory_name, name, 
environment)
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\site-packages\dbx\commands\datafactory.py", line 
91, in __init__
    self.subscription_id = self._get_subscription_id(subscription_name)
  File "C:\Users\lapinska\Anaconda3\envs\dbx_py375\lib\site-packages\dbx\commands\datafactory.py", line 
139, in _get_subscription_id
    sub for sub in self.sub_client.subscriptions.list() if sub.display_name == subscription_name        
AttributeError: 'SubscriptionClient' object has no attribute 'subscriptions'
```

## Reason 
Attribute was renamed from ```subscriptions``` to ```subscription``` in ```azure-mgmt-subscription==2.0.0```
[subscription_client.py#L66](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/subscription/azure-mgmt-subscription/azure/mgmt/subscription/_subscription_client.py#L66)

## Workaround
Fall back to version 1.0.0
```pip install azure-mgmt-subscription==1.0.0```

* dbx version used: DataBricks eXtensions aka dbx, version ~> 0.3.0

**Comment by renardeinside** [comment]
thank you for reporting the bug @alapinsk ! Please feel free to contribute a fix, afaik you've already found the root cause :)

**Comment by renardeinside** [comment]
hi @alapinsk , 

I've upgraded the ADF dependencies bc I also needed to bugfix some other ADF issues. I'll discard your PR since it fixes the old version. Anyway, thanks a lot for spotlighting the issue! Bugfix will be released in the 0.3.1 version.
